subsys/settings/src/settings_init.c

```diff
@@ -117,7 +117,55 @@ int settings_backend_init(void)
 
 	return rc;
 }
-#elif defined(CONFIG_SETTINGS_NONE)
+
+#elif defined(CONFIG_SETTINGS_NVS)
+#include <device.h>
+#include <flash.h>
+#include "settings/settings_nvs.h"
+
+static struct settings_nvs default_settings_nvs;
+
+int settings_backend_init(void)
+{
+	int rc;
+	struct flash_pages_info info;
+
+	rc = flash_get_page_info_by_offs(device_get_binding(DT_FLASH_DEV_NAME),
+					 DT_FLASH_AREA_STORAGE_OFFSET, &info);
+	if (rc) {
+		return rc;
+	}
+	/* define the nvs file system using the page_info */
+	default_settings_nvs.cf_nvs.sector_size =
+		CONFIG_SETTINGS_NVS_SECTOR_SIZE;
+	default_settings_nvs.cf_nvs.sector_count =
+		CONFIG_SETTINGS_NVS_SECTOR_COUNT;
+	default_settings_nvs.cf_nvs.offset = DT_FLASH_AREA_STORAGE_OFFSET +
+		CONFIG_SETTINGS_NVS_OFFSET;
+
+	rc = nvs_init(&default_settings_nvs.cf_nvs, DT_FLASH_DEV_NAME);
```

Comment:
I would move the nvs_init() into the settings_nvs_backend_init(), then a user can do the same procedure without having to use nvs as the default backend.